runtime/src/accounts_db.rs

```diff
@@ -694,9 +699,9 @@ impl AccountsDB {
 
         let mut clean_old_rooted = Measure::start("clean_old_roots");
         if !purges_in_root.is_empty() {
-            self.clean_old_rooted_accounts(purges_in_root);
+            self.clean_old_rooted_accounts(purges_in_root, max_clean_root);
         }
-        self.do_reset_uncleaned_roots(&mut candidates);
+        self.do_reset_uncleaned_roots(&mut candidates, max_clean_root);
```

Comment:
now that `clean_accounts` is out of the latency-sensitive code path of storing accounts. How about shrinking all of newly cleaned accounts? This might fix the snapshot bloat?